src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -260,19 +262,73 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
+        // SetConfig)
+        if (streamsExplicitlySetForModel || streamsExplicitlySetForEngine)
+            return std::string();
+
+        const auto& perf_hint = config.find(CONFIG_KEY(PERFORMANCE_HINT));
+        // the perf_hint may have just arrived to the LoadNetwork, or was set with the plugin's SetConfig
+        if (perf_hint == config.end() && engConfig.perfHintsConfig.ovPerfHint.empty())
+            return std::string();
+        /* performance hints set for network has higher pririty than engine ones.
+         * This applies for all the configuration parameters */
+        const auto perf_hint_name = (perf_hint != config.end())
+                                        ? PerfHintsConfig::CheckPerformanceHintValue(perf_hint->second)
+                                        : engConfig.perfHintsConfig.ovPerfHint;
+        return perf_hint_name;
+    };
+
+    const auto perf_hint_name = getPerfHintName();
+    int streams = engConfig.streamExecutorConfig._streams;
+    if (perf_hint_name == CONFIG_VALUE(LATENCY)) {
+        streams = static_cast<int>(getAvailableNUMANodes().size());
+    } else if (perf_hint_name == CONFIG_VALUE(THROUGHPUT)) {
+        streams = 0;
+    } else if (perf_hint_name.empty()) {
+        streams = streams > 1 ? streams : (engConfig.streamExecutorConfig._set_streams ? streams : 0);
+    }
+
+    auto num_requests = config.find(CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS));
+    int infer_requests = 0;
+    if (num_requests != config.end()) {  // arrived with config to the LoadNetwork (and thus higher pri)
+        infer_requests = PerfHintsConfig::CheckPerformanceHintRequestValue(num_requests->second);
+    } else if (engConfig.perfHintsConfig.ovPerfHintNumRequests) {  // set thru SetConfig to the plugin, 2nd priority
+        infer_requests = engConfig.perfHintsConfig.ovPerfHintNumRequests;
+    }
+
+    const auto common_hints = get_num_streams(streams, infer_requests, ngraphFunc, engConfig.streamExecutorConfig);
+
+    config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = common_hints.first;
+    config[ov::num_streams.name()] = common_hints.first;
+    config[CONFIG_KEY(CPU_THREADS_NUM)] = std::to_string(common_hints.second.num_threads);
+    config[CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)] = std::to_string(common_hints.second.big_core_streams);
+    config[CONFIG_KEY_INTERNAL(BIG_CORE_LOGIC_STREAMS)] = std::to_string(common_hints.second.big_core_logic_streams);
+    config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(common_hints.second.small_core_streams);
+    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] = std::to_string(common_hints.second.threads_per_stream_big);
+    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] =
+        std::to_string(common_hints.second.threads_per_stream_small);
+    config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(common_hints.second.small_core_offset);
+}
```

Comment:
maybe it's better to keep all threading / streams part in a separate file ?
plugin.cpp is a umbrella file which calls all other logic, we should not have implementation of complex logic here

@dmitry-gorokhov what do you think?

Reply:
The problem here is that now we have `GetPerformanceStreams` and `ApplyPerformanceHints` and this is impossible to understand that's the difference. Some refactoring is needed:
1. Lets move `GetPerformanceStreams`, `SetStreamtoConfig` into cpu_streams_caclulation,cpp, . Additionaly lets put  `ApplyPerformanceHints`  and `GetNumStreams` functions is separate file `cpu_streams_caclulation_legacy.cpp`.
2. Lets properly explain why we need to have all this functions in the comments.

Note: This refactoring can be done in separate PR.

@wangleis when we will be able to get rid of legacy approach for streams calculation?

Reply:
Sure, @dmitry-gorokhov. 

Current implementation follow the legacy communication between CPU plugin and StreamsExecutor by using common config_key for minimum change and reusing legacy code. In the following flow refactoring, we may create new communication API for StreamsExecutor only. Then we can pass `streams_info_table` eg.. 

On Windows, new streams calculation can already cover the legacy approach, because new streams calculation is a superset of legacy approach.

On Linux, new streams calculation can already cover normal cases which system expose right cache information. We need more CPU mapping identification method to cover some corner cases, for example

1. Identify CPU mapping based on frequency when cache info is missing in system
2. Estimate CPU mapping as legacy approach based on TBB result when application cannot use all system resource
3. ......

After flow refactoring and CPU mapping identification method are perfected, we can get rid of legacy approach for streams calculation. As we discussed before, it planned in 2023.1 or 2023.2.